### PR TITLE
Publish snapshot on push to develop

### PR DESCRIPTION
**Pull request by adongare**
## MADiE PR Template

Jira Ticket: [MAT-4164](https://jira.cms.gov/browse/MAT-4164)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e. Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: .github/workflows/maven-ci.yml [hunk 1]
```diff
@@ -26,7 +26,14 @@ jobs:
         with:
           distribution: 'adopt'
           java-version: '16'
+
+      - name: Extract project version
+        run: echo ::set-output name=version::$(mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)
+        id: project
+
+      # publish snapshot iff the project version in pom.xml contains '-SNAPSHOT'
```
Comment (RohitKandimalla) [inline .github/workflows/maven-ci.yml:34]:
```suggestion
      # publish snapshot if the project version in pom.xml contains '-SNAPSHOT'
```

**Review by RohitKandimalla — APPROVED** [review]

File: .github/workflows/maven-ci.yml (hunk 1 above)
Comment (adongare) [reply .github/workflows/maven-ci.yml:34]: it was like `if and only if `
